server/src/auth/token_validator.rs

```diff
@@ -18,6 +18,21 @@ pub struct TokenValidator {
     pub persistence: Option<Arc<dyn EdgePersistence>>,
 }
 
+pub(crate) trait TokenRegister {
```

Comment:
I've carved this method out into its own trait so we can make the validation middleware testable, otherwise its exactly the same logic